### PR TITLE
HDDS-5524. Use try-with-resources to properly close resources in DirectoryServerSource.java

### PR DESCRIPTION
**Pull request by sky76093016**
## What changes were proposed in this pull request?

Resources should be closed.Failure to properly close resources will result in a resource leak.
Resource should be created using `try-with-resources` pattern and will be closed automatically.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/projects/HDDS/issues/HDDS-5524

## How was this patch tested?

No need.

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/stream/DirectoryServerSource.java [hunk 1]
```diff
@@ -41,18 +42,19 @@ public DirectoryServerSource(Path root) {
...
+          throws InterruptedException {
     Map<String, Path> files = new HashMap<>();
     final Path streamingDir = root.resolve(id);
     try {
-      Files.walk(streamingDir)
-          .filter(Files::isRegularFile)
-          .forEach(path -> {
-            files.put(root.relativize(path).toString(), path);
-          });
+      try (Stream<Path> list = Files.walk(streamingDir)
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/stream/DirectoryServerSource.java:49]: Can you please merge these two `try` blocks?

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/stream/DirectoryServerSource.java [hunk 2]
```diff
@@ -41,18 +42,19 @@ public DirectoryServerSource(Path root) {
    * @param id name of the subdirectory to replitace relative to root.
    */
   public Map<String, Path> getFilesToStream(String id)
-      throws InterruptedException {
+          throws InterruptedException {
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/stream/DirectoryServerSource.java:45]: Nit: unnecessary whitespace change.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/stream/DirectoryServerSource.java [hunk 3]
```diff
@@ -41,18 +42,19 @@ public DirectoryServerSource(Path root) {
...
+      try (Stream<Path> list = Files.walk(streamingDir)
+              .filter(Files::isRegularFile)) {
+        list.forEach(path -> {
+          files.put(root.relativize(path).toString(), path);
+        });
+      }
     } catch (IOException e) {
       throw new StreamingException("Couldn't read directory for streaming: " +
-          streamingDir, e);
+              streamingDir, e);
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/stream/DirectoryServerSource.java:57]: Nit: unnecessary whitespace change.

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @sky76093016 for the fix.  Minor code improvement suggested.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/stream/DirectoryServerSource.java (hunk 1 above)
Comment (sky76093016) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/stream/DirectoryServerSource.java]: Thanks @adoroszlai for reviewing, I have fixed it.

**Review by adoroszlai — APPROVED** [review]
Thanks @sky76093016 for updating the patch.

**Review by cku328 — APPROVED** [review]
Thanks @sky76093016 for fixing it.
I think there are other files that have similar issues that should be fixed as well.

**Comment by sky76093016** [comment]
Thanks @cku328 for reviewing,I will fix those issues in the future.
